magicbyirineu/Modules/CardDetail/View/CardDetailViewController.swift

```diff
@@ -49,4 +50,13 @@ class CardDetailViewController: MagicViewController {
         self.dismiss(animated: true, completion: nil)
     }
     
+    @objc func favoriteButtonTapped(){
+        switch self.screen.favoriteButton.isFavorite {
+        case true:
+            self.screen.favoriteButton.setAsNotFavorite()
```

Comment:
Esse tipo de expressão foge a regra do `Law Of Demeter`

Vide [https://pt.wikipedia.org/wiki/Lei_de_Demeter#Vantagens](url).

Sugiro ter um método na screen que faça isso.